salt/react_apps.sls

```diff
@@ -54,39 +55,47 @@ include:
     - cwd: {{ directory }}
     - require:
       - pkg: git
+    - onchanges:
+      - git: {{ entry.git.url }}
 
-{% if enable_ver_txt %}
-
-{{ build_dir }}/ver.txt:
+{% if salt['pillar.get']('ver_txt:enable') %}
+{{ build_asset_dir }}/ver.txt:
   file.managed:
-    - contents: "branch: {{ entry.git.branch }} || commit_hash: {{ salt['cmd.shell']('cd '+ directory +' && git rev-parse --verify HEAD') }} || time: {{ tstamp }}"
-
+    - contents: "branch: {{ entry.git.branch }} || commit_hash: {{ salt['cmd.shell']('cd ' + directory + '&& git rev-parse --verify HEAD') }} || time: {{ timestamp }}"
+    - onchanges:
+      - cmd: {{ userdir }}-yarn-build
 {% endif %}{# ver txt #}
 
 {{ directory }}/.htaccess:
   file.managed:
-    - name: {{ build_dir }}/.htaccess
+    - name: {{ build_asset_dir }}/.htaccess
     - source: {{ directory }}/.htaccess
+    - onchanges:
+      - cmd: {{ userdir }}-yarn-build
```

Comment:
For the `file` states, `onchanges` means they will only run if the `cmd` states run (which in turn only run if git changed).

This state copies a file from git, so it should work in normal circumstances. However, in this case, the change in the `name` won't be deployed unless `git` changed.

An alternative to to add a state:

```
{{ build_asset_dir }}:
  file.exists
```

Then, in the `file` states, use:

```
    - require:
      - file: {{ build_asset_dir }}
```

That way, they won't run if the directory doesn't exist.

The `file.exists` state will need to be after the `cmd` states.